examples/jsm/nodes/functions/PhysicalLightingModel.js

```diff
@@ -28,14 +32,53 @@ const computeMultiscattering = ( singleScatter, multiScatter, specularF90 = floa
 
 };
 
-const RE_IndirectSpecular_Physical = new ShaderNode( ( inputs ) => {
+const LM_Init = new ShaderNode( ( context, stack, builder ) => {
+
+	if ( builder.includes( clearcoat ) ) {
+
+		context.clearcoatRadiance = vec3().temp();
+		context.reflectedLight.clearcoatSpecular = vec3().temp();
+
+		const dotNVcc = transformedClearcoatNormalView.dot( positionViewDirection ).clamp();
+
+		const Fcc = F_Schlick.call( {
+			dotVH: dotNVcc,
+			f0: clearcoatF0,
+			f90: clearcoatF90
+		} );
+
+		const outgoingLight = context.reflectedLight.total;
+		const clearcoatLight = outgoingLight.mul( clearcoat.mul( Fcc ).oneMinus() ).add( context.reflectedLight.clearcoatSpecular.mul( clearcoat ) );
+
+		outgoingLight.assign( clearcoatLight );
+
+	}
+
+} );
+
+const RE_IndirectSpecular_Physical = new ShaderNode( ( context ) => {
 
-	const { radiance, iblIrradiance, reflectedLight } = inputs;
+	const { radiance, iblIrradiance, reflectedLight } = context;
+
+	if ( reflectedLight.clearcoatSpecular ) {
+
+		const dotNVcc = transformedClearcoatNormalView.dot( positionViewDirection ).clamp();
+
+		const clearcoatEnv = EnvironmentBRDF.call( {
+			dotNV: dotNVcc,
+			specularColor: clearcoatF0,
+			specularF90: clearcoatF90,
+			roughness: clearcoatRoughness
+		} );
+
+		reflectedLight.clearcoatSpecular.addAssign( context.clearcoatRadiance.mul( clearcoatEnv ) );
+
+	}
 
 	// Both indirect specular and indirect diffuse light accumulate here
 
-	const singleScattering = temp( vec3() );
-	const multiScattering = temp( vec3() );
+	const singleScattering = vec3().temp();
```

Comment:
What is `temp()`? 🤔

Reply:
It allows to create a new temporary variable.

Reply:
would `clone()` work too?

Reply:
No. I meant "variable" as not a JS variable, but a variable in the final GLSL/WGSL code.

Reply:
`temp` just tells to explicitly create a new variable, so that we can do assigning to it.

Reply:
The Node System tries to optimize the code automatically putting all code in a single line if possible. If a node is used more than two time they automatically create a temporary variable. Like:

```wgsl
vec3<f32> nodeVar2 = ...
```

When  `.temp()` is used force the system to create a variable, it is useful to store a variable that the user can modify itself using `addAssign()`, `subAssign()`, or `assign()` for example. This syntax in WGSL is like:

```wgsl
// TSL: const node = vec3().temp()
vec3<f32> nodeVar2 = vec3<f32>( 0.0 );

// TSL: node.addAssign( 1.0 )
nodeVar2 = nodeVar2 + vec3<f32>( 1.0 );
```

```js
// without temp
let node = vec3();
node = node.add( 1.0 );
// WGLSL: ... ?= ( vec3<f32>( 0.0 ) +  vec3<f32>( 1.0 ) )

// using temp
const node = vec3().temp();
node.addAssign( 1.0 );

// WGLSL:
// vec3<f32> nodeVar2 = vec3<f32>( 0.0 );
// nodeVar2 = nodeVar2 + vec3<f32>( 1.0 );
```

Without temp is used in most of the time because have a better syntax using chain but in some cases when an incremental variable is used like `outgoingLight` is called in differences functions seems better approach to avoid use async of variable.

Reply:
Got it. Thanks!